sklearn/impute.py

```diff
@@ -249,7 +250,8 @@ def _sparse_fit(self, X, strategy, missing_values, axis):
 
     def _dense_fit(self, X, strategy, missing_values, axis):
         """Fit the transformer on dense data."""
-        X = check_array(X, force_all_finite=False)
+        X = check_array(X, force_all_finite='allow-nan'
+                        if self.missing_values == 'NaN' else True)
```

Comment:
`self.missing_values == 'NaN' or np.isnan(self.missing_values)`? (like what we do in _get_mask)

Reply:
Good idea thanks.